### PR TITLE
Fixreboot

### PR DESCRIPTION
**Pull request by aojea**

### CODE REVIEW

File: images/base/files/usr/local/bin/entrypoint [hunk 1]
```diff
@@ -181,13 +181,28 @@ enable_network_magic(){
   sed -e "s/${docker_embedded_dns_ip}/${docker_host_ip}/g" /etc/resolv.conf.original >/etc/resolv.conf
 
   # fixup IPs in manifests ...
-  # TODO: ensure this supports rebooting the host w/ IPv6 clusters
-  curr_ip=$(hostname --ip-address)
-  if [ -f /kind/old-ip ]; then
-    old_ip=$(cat /kind/old-ip)
-    sed -i "s#${old_ip}#${curr_ip}#" /etc/kubernetes/manifests/*.yaml /var/lib/kubelet/kubeadm-flags.env
+  curr_ipv4="$( (getent ahostsv4 $(hostname) | head -n1 | cut -d' ' -f1) || true)"
```
Comment (aojea) [inline images/base/files/usr/local/bin/entrypoint:184]: I've tried to put it in an array but is too brittle, this option has more boilerplate code bit is more resilient ... the order of the returned address may change depending on the node configuration, or someone can enable ipv6 addresses later on, who knows :shrug:

**Comment by aojea** [comment]
@BenTheElder please take the last commit , it fixes the reboot on ipv6 and in worker nodes

**Comment by aojea** [comment]
I've tested and it worked to me on Linux, IPv6 and multinode, can't test in mac or windows

File: images/base/files/usr/local/bin/entrypoint (hunk 1 above)
Comment (BenTheElder) [reply images/base/files/usr/local/bin/entrypoint:184]: if we depended on this for something other than host reboot I'd be pretty unhappy with how much `|| true` is going on here, but given the context this seems reasonable..

**Comment by BenTheElder** [comment]
xref: #148 #1475

**Comment by BenTheElder** [comment]
rebased, droped the old base bump, added a new base bump

**Comment by BenTheElder** [comment]
/lgtm cancel
/approve cancel
/hold
this needs to be reworked, this is not safe when there is not an IPv6 for one.

File: images/base/files/usr/local/bin/entrypoint [hunk 2]
```diff
@@ -181,13 +181,28 @@ enable_network_magic(){
...
+  echo "Current ip $curr_ipv4"
+  if [ -f /kind/old-ipv4 ]; then
+      old_ipv4=$(cat /kind/old-ipv4)
+      # kubernetes manifests are only present on control-plane nodes
+      sed -i "s#${old_ipv4}#${curr_ipv4}#" /etc/kubernetes/manifests/*.yaml || true
+      # this is no longer required with autodiscovery
+      sed -i "s#${old_ipv4}#${curr_ipv4}#" /var/lib/kubelet/kubeadm-flags.env || true
   fi
-  echo -n "${curr_ip}" >/kind/old-ip
+  [[ ! -z "${curr_ipv4}" ]] && echo -n "${curr_ipv4}" >/kind/old-ipv4
```
Comment (BenTheElder) [inline images/base/files/usr/local/bin/entrypoint:193]: this form is cute but exits. an if statement does not set the exit code.

**Comment by BenTheElder** [comment]
fixing

File: images/base/files/usr/local/bin/entrypoint (hunk 1 above)
Comment (BenTheElder) [reply images/base/files/usr/local/bin/entrypoint:184]: guarding against this more in another commit

**Comment by BenTheElder** [comment]
pushing a new node image because this change will otherwise break the default :-)

**Comment by aojea** [comment]
Is still on hold , any particular reason?

**Pull request merged**